File: src/main/java/rooms/IRoom.java
package rooms;

import people.male.IMan;
import people.IPerson;
import people.female.IWoman;

import java.util.ArrayList;

public interface IRoom {
    String getName();
    IRoom setName(String name);
    IRoom addWoman(String firstName, String lastName);
    IRoom addMan(String firstName, String lastName);
    ArrayList<IPerson> getPeople();
    ArrayList<IWoman> getWomen();
    ArrayList<IMan> getMen();
}

File: src/kotlin/KotlinExamples/src/IRoom.kt
import java.util.ArrayList

interface IRoom {
    var name: String?
    var people: ArrayList<IPerson>?
    var women: ArrayList<IWoman>?
    var men: ArrayList<IMan>?
    fun setName(name: String): IRoom
    fun addPerson(person: IPerson): IRoom
    fun addWoman(firstName: String, lastName: String): IRoom
    fun addMan(firstName: String, lastName: String): IRoom
}